src/Kunstmaan/NodeBundle/EventListener/NodeTranslationListener.php

```diff
@@ -324,7 +325,18 @@ private function ensureUniqueUrl(
             ).' node(s) that match url \''.$translation->getUrl().'\''
         );
 
-        if (count($translations) > 0) {
+        $translationsWithSameUrl = array();
+
+        /** @var NodeTranslation $trans */
+        foreach ($translations as $trans) {
+            $refEntityName = $trans->getPublicNodeVersion()->getRefEntityName();
+            $object = new $refEntityName();
+            if (!$object instanceof StructureNode) {
```

Comment:
How is even `StructureNode` any special here? This is an implementation detail. I have structure nodes/pages that do not extend this class. `instanceof StructureNode` does not determine if a page is a structure node. `HasNodeInterface::isStructureNode()` does.

Also in this commit:

 * Creating a page assuming it has no constructor arguments. Poor design
 * Checking if instance implements interface / extends class, instead of checking if class extends / implements (using `class_extends` / `class_implements`). This would fix the first issue.
 * Reimplementing business logic. Please use `PagesConfiguration::isStructureNode($refNameOrInstance)` for this